src/components/Provider/BreakpointProvider.tsx

```diff
@@ -0,0 +1,51 @@
+import debounce from 'lodash/debounce';
+import * as React from 'react';
+import {
+	Breakpoint,
+	BreakpointContext,
+} from '../../contexts/BreakpointContext';
+export interface BreakpointProviderProps {
+	breakpoints: number[];
+	onBreakpointChange?: (breakpoint: Breakpoint) => void;
+}
+
+export const BreakpointProvider: React.FC<BreakpointProviderProps> = ({
+	breakpoints,
+	children,
+}) => {
+	const [currentBreakpoint, setCurrentBreakpoint] = React.useState(
+		0 as Breakpoint,
+	);
+
+	const handleResize = () => {
+		const clientWidth = window.innerWidth;
+		if (clientWidth < breakpoints[0]) {
+			setCurrentBreakpoint(0);
+		} else if (clientWidth < breakpoints[1]) {
+			setCurrentBreakpoint(1);
+		} else if (clientWidth < breakpoints[2]) {
+			setCurrentBreakpoint(2);
+		} else if (clientWidth >= breakpoints[2]) {
+			setCurrentBreakpoint(3);
+		}
+	};
+
+	React.useEffect(() => {
+		const debouncedResize = debounce(handleResize, 80);
```

Comment:
Once we move handleResize in here and make it dynamic, before any of this is called, we should validate that `breakpoints` exists and that the length is not 0 (and maybe set some maximum as well).